redshift/resource_redshift_group.go

```diff
@@ -213,6 +213,34 @@ func setGroupName(tx *sql.Tx, d *schema.ResourceData) error {
 	return nil
 }
 
+func checkIfUserExists(tx *sql.Tx, name interface{}) (bool, error) {
```

Comment:
I highly recommend making `name` a `string` type instead of an `interface` type. You will avoid casting and checking whether it was successful.

Reply:
No, you will no avoid casting. When you check setUsersNames function and for cycle inside you will see that List() returns list of interfaces and name is interface.
`for _, name := range removedUsers.List() {`

Reply:
updated

Reply:
What i meant is that you can change the function signature to be `func checkIfUserExists(tx *sql.Tx, name string) (bool, error)` and move the casting to `setUsersNames` function before calling `checkIfUserExists`.

`removedUsers` is a set created from the schema attribute `users`. The reason why you would not need to check if casting to string was successful is that the terraform schema already did that for you. The input you receive is already validated and you can safely assume it is of a type defined in the schema (in this case, a string)